pyproject.toml

```diff
@@ -38,3 +38,49 @@ exclude = '''
   | asv_benchmarks/env
 )/
 '''
+
+[tool.cython-lint]
+# Ignore the same error codes as flake8
+# + E501 (line too long) because keeping it < 88 in cython
+# often makes code less readable.
+ignore = [
+    # check ignored by default in flake8. Meaning unclear.
+    'E24',
+    # space before : (needed for how black formats slicing)
+    'E203',
+    # line too long
+    'E501',
+    # do not assign a lambda expression, use a def
+    'E731',
+    # do not use variables named 'l', 'O', or 'I'
+    'E741',
+    # line break before binary operator
+    'W503',
+    # line break after binary operator
+    'W504',
+]
+# Exclude files are generated from tempita templates
+exclude= '''
+(
+    sklearn/_loss/_loss.pyx
+  | sklearn/linear_model/_sag_fast.pyx
+  | sklearn/linear_model/_sgd_fast.pyx
+  | sklearn/utils/_seq_dataset.pyx
+  | sklearn/utils/_seq_dataset.pxd
+  | sklearn/utils/_weight_vector.pyx
+  | sklearn/utils/_weight_vector.pxd
+  | sklearn/metrics/_dist_metrics.pyx
+  | sklearn/metrics/_dist_metrics.pxd
+  | sklearn/metrics/_pairwise_distances_reduction/_argkmin.pxd
+  | sklearn/metrics/_pairwise_distances_reduction/_argkmin.pyx
+  | sklearn/metrics/_pairwise_distances_reduction/_argkmin_classmode.pyx
+  | sklearn/metrics/_pairwise_distances_reduction/_base.pxd
+  | sklearn/metrics/_pairwise_distances_reduction/_base.pyx
+  | sklearn/metrics/_pairwise_distances_reduction/_datasets_pair.pxd
+  | sklearn/metrics/_pairwise_distances_reduction/_datasets_pair.pyx
+  | sklearn/metrics/_pairwise_distances_reduction/_middle_term_computer.pxd
+  | sklearn/metrics/_pairwise_distances_reduction/_middle_term_computer.pyx
+  | sklearn/metrics/_pairwise_distances_reduction/_radius_neighbors.pxd
+  | sklearn/metrics/_pairwise_distances_reduction/_radius_neighbors.pyx
+)
+'''
```

Comment:
What about using `git ls-files` as suggested in https://github.com/scikit-learn/scikit-learn/pull/25861#issuecomment-1488412855 instead of maintaining a materialized exclusion list?

I am fine with both approaches, but maybe the `git ls-files` trick would be lower maintenance?

Reply:
`git ls-files` is fine in the CI but I wanted that contributors can just run `cython-lint .` like they run `black` or `flake8`.

Reply:
We now have three lists to maintain (this one, the one in `setup.py`, and the one in `setup.cfg`). Is there a way to have fewer lists to maintain?

Reply:
`pre-commit` would solve this for you, as it automatically handles parallelisation + only running on tracked files ;) pandas uses it for everything linting/formatting-related, can recommend

otherwise, I guess you could make a local script `run_cython_lint.py` which runs something like
```python
files = subprocess.run(['git ls-files'], text=True, capture_output=True).strip().splitlines()
subprocess.run(['cython-lint', *files])
```
and then both contributors and CI run `python scripts/run_cython_lint.py`?

Or write another script to keep the three lists in sync and have it run in CI? This might be smallest-effort, least disruptive solution?

Reply:
> pre-commit would solve this for you

I don't want to force our contributors to use `pre-commit`. I don't use it for instance :)

>  then both contributors and CI run python scripts/run_cython_lint.py?

I like the idea of contributors being able to invoke the cython linter the same way they invoke the other linters. We have a list of excluded directories/files for black in `pyproject.toml`. I think it's acceptable to have a list for `cython-lint` as well.